cmd/oras/root/push.go

```diff
@@ -224,25 +231,62 @@ func runPush(ctx context.Context, opts pushOptions) error {
 	return opts.ExportManifest(ctx, memoryStore, root)
 }
 
-func updateDisplayOption(opts *oras.CopyGraphOptions, fetcher content.Fetcher, verbose bool) {
+func doPush(dst oras.Target, pack packFunc, copy copyFunc) (ocispec.Descriptor, error) {
+	if tracked, ok := dst.(track.GraphTarget); ok {
+		defer tracked.Close()
+	}
+	// Push
+	return pushArtifact(dst, pack, copy)
+}
+
+func updateDisplayOption(opts *oras.CopyGraphOptions, fetcher content.Fetcher, verbose bool, tracked track.GraphTarget) {
 	committed := &sync.Map{}
-	opts.PreCopy = display.StatusPrinter("Uploading", verbose)
 	opts.OnCopySkipped = func(ctx context.Context, desc ocispec.Descriptor) error {
 		committed.Store(desc.Digest.String(), desc.Annotations[ocispec.AnnotationTitle])
-		return display.PrintStatus(desc, "Exists   ", verbose)
+		if tracked == nil {
+			return display.PrintStatus(desc, "Exists  ", verbose)
+		}
+		return tracked.Prompt(desc, "Exists   ", verbose)
+	}
```

Comment:
nit: how about putting this block of code to the if statement at line 251 and after line 271? just like what you have done to `opts.PostCopy`?